### DISCUSSION TITLE
Should we invest in a PyO3 chat space?

### DISCUSSION THREAD

**Discussion by davidhewitt** [PyO3/pyo3#3857] [Polls]
EDIT: it's decided, we will use Discord! Join us at https://discord.gg/c4BwayXQ

---- 

As I wrote in the "long term" section of [my recent opening blog post](https://polar.sh/davidhewitt/posts/hello-world-and-the-future-for-pyo3), I've been thinking a lot recently about how to bring the PyO3 community together.

GitHub is great for slow asynchronous communication. It's very formal and searchable.

I think there's quite a bit to tap from investing in a community space with less formal and faster communication. I wrote quite a bit about the community opportunities in my blog so I won't repeat those here. There are also cases like https://github.com/PyO3/pyo3/issues/3684#issuecomment-1925781502 where it might be helpful for maintainers to easily ping each other to say what they're working on & when.

My main two problems I've had with our existing Gitter channel which have held me back from investing in it:
 - UX pains with Matrix/Gitter (which might be mostly resolved now)
 - It's a single channel. I'd like a space with multiple channels for different topics. I think this would help with separating topics like user questions and "show and tell" from PyO3's own development chatter.

I am currently feeling quite tempted to give Discord a serious go as a new PyO3 chat space. Obviously there is no point creating a chat space if I'm the only one in it, however! Hence my question.

Frequency of your participation is not important. I think this will naturally ebb and flow for everyone. As long as there is enough people in the chat space that questions are answered eventually, I think the space can self sustain.

**Comment by davidhewitt** [comment]
The fact that this response is not a unanimous "yes" highlights an important point I want to make: it is very important in my opinion that the PyO3 design process remains on GitHub. There is no need to move design prototyping and decisions and we would lose so much from making them in a less searchable medium.

An informal chat space still has much to offer without being the place where PyO3 is designed. We should try very very hard to encourage discussion to move to GitHub issue whenever features ideas and bug reports become the chat subject; this is already what I do on the Gitter chat room.

**Comment by LilyFirefly** [comment]
Either of the Discord options would work for me. I think our own server makes a little more sense, which is why I voted that one. I have no experience with Matrix or Gitter, so have no real opinion about them.

**Comment by mejrs** [comment]
Our own server makers the most sense to me. I'll be there :)

**Comment by davidhewitt** [comment]
As the response seems to generally be quite positive towards "yes, let's create a Discord server", I went ahead and did it.

~~Invite link: https://discord.gg/c4BwayXQ~~

I think for a couple days let's just see what we think of the platform and agree how we want the server set up (I made a meta channel on the server where we can discuss that). If we're happy and decide after that time that we'd like to redirect users from Gitter to Discord, we can update the README and put some messaging on the Gitter room.

**Reply by davidhewitt** [reply]
EDIT: updated link https://discord.gg/33kcChzH7f (the other one expired, my mistake).

**Comment by k82cn** [comment]
Discord sounds good :)
